3.1 Factor code into three files

mpharrigan opened the pull request:
This PR is purely moving things around (on top of https://github.com/quantumlib/Cirq/pull/3760/) with one exception noted inline.

 - `random_quantum_circuit_generation` -- left alone; this is a well-scoped file
 - `xeb_sampling` - functionality related to sampling random circuits efficiently (for xeb)
 - `xeb_simulation` - functionality related to simulating 2-qubit circuits at various depths (for xeb) efficiently using multiprocessing
 - `xeb_fitting` - functionality related to using xeb to characterize gates by fitting parameters.

mpharrigan committed:
[XEB] Move things

mpharrigan commented on cirq/experiments/xeb_fitting.py:
```diff
@@ -0,0 +1,298 @@
+):
+    """Simulate and benchmark two-qubit XEB circuits.
+
+    This uses the estimator from
+    `cirq.experiments.fidelity_estimation.least_squares_xeb_fidelity_from_expectations`, but
+    adapted for use on pandas DataFrames for efficient vectorized operation.
```
This is new

mpharrigan committed:
Move tests

mpharrigan committed:
Notebook new imports

balopat commented:
> This PR is purely moving things around (on top of [quantumlib#3760](https://github.com/quantumlib/Cirq/pull/3760)) with one exception noted inline.
> 
> * `random_quantum_circuit_generation` -- left alone; this is a well-scoped file
> * `xeb_sampling` - functionality related to sampling random circuits efficiently (for xeb)
> * `xeb_simulation` - functionality related to simulating 2-qubit circuits at various depths (for xeb) efficiently using multiprocessing
> * `xeb_fitting` - functionality related to using xeb to characterize gates by fitting parameters.

This organization makes sense to me, no objections!

mpharrigan committed:
Merge branch '2021-01-xeb-3' into 2021-01-xeb-3.1-move

mpharrigan committed:
[XEB] Split into three files

mpharrigan committed:
Merge branch '2021-02-xeb-3.1-move' into 2021-01-xeb-3.1-move

balopat committed:
Notebooks guide (#3781)

Adds guidance on styling and development of notebooks.

mpharrigan committed:
Merge branch 'master' into 2021-02-xeb-3.1-move

mpharrigan committed:
[XEB] Split into three files (#3794)

This PR is purely moving things around (on top of https://github.com/quantumlib/Cirq/pull/3760/) with one exception noted inline.

 - `random_quantum_circuit_generation` -- left alone; this is a well-scoped file
 - `xeb_sampling` - functionality related to sampling random circuits efficiently (for xeb)
 - `xeb_simulation` - functionality related to simulating 2-qubit circuits at various depths (for xeb) efficiently using multiprocessing
 - `xeb_fitting` - functionality related to using xeb to characterize gates by fitting parameters.

mpharrigan committed:
Merge branch '2021-02-xeb-3.1-move' into 2021-01-xeb-3.1-move

mpharrigan committed:
Merge remote-tracking branch 'origin/master' into 2021-01-xeb-3.1-move
